crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -93,16 +100,15 @@ impl Trade {
     /// fees have been applied.
     ///
     /// Denominated in SURPLUS token
-    fn surplus(&self) -> Option<eth::Asset> {
+    fn surplus_over_limit_price(&self, limit: Amounts) -> Option<eth::Asset> {
```

Comment:
This change also requires to update the comment above the function, so that the client is aware that surplus is calculated over provided limit amounts.